Python/PyHSPlasma.pyi

```diff
@@ -661,7 +663,10 @@ class hsKeyFrame:
     def write(self, stream: hsStream) -> None: ...
 
 class hsKeyedObject(plCreatable):
-    key: plKey = ...
+    if sys.version_info.major == 3 and sys.verion_info.minor < 11:
+        key: plKey = ...
+    else:
+        key: plKey[Self] = ...
```

Comment:
I know I'm late, but the Python version check can be written shorter (and more correctly) as:

```suggestion
    if sys.version_info < (3, 11):
        key: plKey = ...
    else:
        key: plKey[Self] = ...
```

And I *think* you can even make this fully compatible with older Python versions if you write it using `@property` and a regular type variable:

```suggestion
    @property
    def key(self: T) -> plKey[T]: ...
```

Reply:
This LGTM. I think you should open this as a pull request 👍